terraform/root/aws-dm-dev/main.tf

```diff
@@ -8,3 +8,9 @@ module "aws_env" {
   aws_main_account_id = "${var.aws_main_account_id}"
   aws_dev_account_id = "${var.aws_dev_account_id}"
 }
+
+module "s3-bucket" {
```

Comment:
If the module is not reused then I would say it's even better if you just define it directly in aws-dm-dev.